### PR TITLE
docs: add documentation for callback arguments

### LINKED ISSUE #628: list what arguments go into the callback
I think this is actually a more widely spread problem, where the examples have been saving us, but...

Is there any way we could more broadly document the arguments that go into the callback function?

### PR DESCRIPTION
**Pull request by callmehiphop**
Fixes #628 
- [x] Document all the things!
  - [x] BigQuery
  - [x] Datastore
  - [x] PubSub
  - [x] Search
  - [x] Storage

### CODE REVIEW

File: lib/bigquery/dataset.js [hunk 1]
```diff
@@ -64,6 +64,9 @@ function Dataset(bigQuery, id) {
  *     JSON array of fields, which allows for nested and repeated fields. See
  *     a [Table resource](http://goo.gl/sl8Dmg) for more detailed information.
  * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request
+ * @param {module:bigquery/table} callback.table - The newly created table.
```
Comment (stephenplusplus) [inline lib/bigquery/dataset.js]: I'm not sure on the right approach here, as far as if this is `nullable` or not. Technically, the callback will get the `table` argument only in non-error cases, so should `{module:bigquery/table}` be `{?module:bigquery/table}`? The only constant, never-null argument is `callback.apiResponse`.

I don't know if that's just going to make things confusing by being technically correct to that detail. Any thoughts?

Comment (callmehiphop) [inline lib/bigquery/dataset.js]: Maybe neither should `nullable`, I think anyone familiar with node will understand the convention

Comment (stephenplusplus) [inline lib/bigquery/dataset.js]: True, though I think at least `err` should be. I'm cool with leaving the second argument not marked as nullable. I think that would just create a lot of noise on the site.

Comment (callmehiphop) [inline lib/bigquery/dataset.js]: I saw nothing :sunglasses:

**Comment by stephenplusplus** [comment]
This is a huge step towards even more docs-awesomeness. Thanks, sir!

**Comment by callmehiphop** [comment]
:dancer:

**Comment by jgeewax** [comment]
Yes-- huge +1 @callmehiphop  :-D

**Pull request merged**